ministryofjustice/parliamentary-questions
# Upgrade Rails to v6.1.3.1 & Ruby to v2.7.2 (CT-3549 + CT-3562)

## Description

- Upgrade Ruby to v2.7.2
- Upgrade Rails to v6.1.3.1
- Sentry setup
- sentry-rails and sentry-ruby gem upgrades 
- Fix deprecated calls to:
	Render actions with '.' in the name.
	Call `<<` to an ActiveModel::Errors.
- Add config/storage.yml file
- Rubocop updates

## Self-review checklist



### Related JIRA tickets

CT-3487 Upgrade activerecord gem to version 6.0.3.5 or later
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-3487

CT-3488 Upgrade actionpack gem to version 6.0.3.5 or later
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-3488

CT-3494 Set up sentry on PQ
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-3494

CT-3549 Upgrade to ruby 2.7.x
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&selectedIssue=CT-3549

CT-3562 Update Rails to v6.1.3.1
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-3562

CT-3571 Update PQ Rails 6.1.3.1 (MimeMagic)
https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-3571

### Manual testing instructions

On the Staging environment:
Test sending/receiving emails related to PQ assignment, PQ acceptance and PQ rejection cycles.
Manual question progression.
Reports production.
Settings - add / edit / delete users / directors etc.
Finally test the happy path to see if anything looks unusual.
